WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/AnalogWatchCanvasRenderer.kt

```diff
@@ -0,0 +1,546 @@
+    // class.
+    private val watchFaceDataObserver: Observer<AnalogWatchFaceAndStylesAndDimensions> =
+        Observer<AnalogWatchFaceAndStylesAndDimensions> {
+            val previousMinuteLength =
+                analogWatchFaceAndStylesAndDimensions?.minuteHandDimensions?.lengthFraction ?: 0.0f
+
+            val newMinuteLength = it.minuteHandDimensions.lengthFraction
+
+            analogWatchFaceAndStylesAndDimensions = it
+
+            if (previousMinuteLength != newMinuteLength) {
+                armLengthChangedRecalculateClockHands = true
+            }
+        }
+
+    // Changed when setting changes cause a change in the minute hand arm (triggered by user in
+    // updateUserStyle() via userStyleRepository.addUserStyleListener()).
+    private var armLengthChangedRecalculateClockHands: Boolean = false
+
+    // Default key assigned for this watch face.
+    private val analogWatchFaceKeyId =
+        com.example.android.wearable.alpha.data.db.analogWatchFaceKeyId
+
+    // Contains all data to render this analog watch face (combines multiple tables).
+    private var analogWatchFaceAndStylesAndDimensions: AnalogWatchFaceAndStylesAndDimensions? = null
+
+    // Default size of watch face drawing area, that is, a no size rectangle. Will be replaced with
+    // valid dimensions from the system.
+    private var currentWatchFaceSize: Rect = Rect(0, 0, 0, 0)
+
+    // Initializes paint object for painting the clock hands with default values.
+    private val clockHandPaint = Paint().apply {
+        isAntiAlias = true
+        strokeWidth = context.resources.getDimensionPixelSize(R.dimen.clock_hand_stroke_width).toFloat()
+    }
+
+    private val outerElementPaint = Paint().apply {
+        isAntiAlias = true
+    }
+
+    // Used to paint the main hour hand text with the hour pips, i.e., 3, 6, 9, and 12 o'clock.
+    private val textPaint = Paint().apply {
+        isAntiAlias = true
+        textSize = context.resources.getDimensionPixelSize(R.dimen.hour_mark_size).toFloat()
+    }
+
+    private lateinit var hourHandFill: Path
+    private lateinit var hourHandBorder: Path
+    private lateinit var minuteHandFill: Path
+    private lateinit var minuteHandBorder: Path
+    private lateinit var secondHand: Path
+
+    init {
+        // It's possible for the user to change the minute arm hand length via the settings, so
+        // we need to observe any changes from that particular table [] and then render them.
+        analogWatchFaceViewModel.getAnalogWatchFaceAndStylesAndDimensions(analogWatchFaceKeyId)
+            .observeForever(watchFaceDataObserver)
+
+        userStyleRepository.addUserStyleListener(
+            object : UserStyleRepository.UserStyleListener {
+                @SuppressLint("SyntheticAccessor")
+                override fun onUserStyleChanged(userStyle: UserStyle) {
+                    Log.d(TAG, "onUserStyleChanged(), userStyle: \n${userStyle.toMap()}")
+                    scope.launch {
```

Comment:
NB if you make scope internal you avoid the SuppressLint.

Reply:
Thanks for the heads up, I didn't realize that.

I actually moved this coroutine that updates settings to DB to view model, so I removed the scope all together, but I remove anyway. :)